engine-precompiles/src/modexp.rs

```diff
@@ -507,4 +513,52 @@ mod tests {
         );
         assert_eq!(Err(ExitError::OutOfGas), res);
     }
+
+    #[test]
+    fn test_modexp_not_overflow() {
+        let base_len = U256::from(0);
+        let exp_len = U256::from(4294967200usize); // `usize::MAX` - 95
+        let mod_len = U256::from(0);
+        let base = U256::MAX;
+        let exp = U256::MAX;
+        let modulus = U256::MAX;
+
+        let mut input: Vec<u8> = Vec::new();
+        input.extend_from_slice(&u256_to_arr(&base_len));
```

Comment:
You can use here `.extend` method.
```suggestion
        input.extend(u256_to_arr(&base_len));
```